Controller/src/controller/controller.go

```diff
@@ -23,65 +23,73 @@ package controller
 import (
 	"log"
 
-	"github.com/FirebaseExtended/fcm-external-prober/Probe/src/utils"
 	"github.com/golang/protobuf/proto"
 )
 
-var maker utils.CommandMaker
+var (
+	maker          utils.CommandMaker
+	vms            map[string]*regionalVM
+	stoppedVMs     int32
+	stoppedVMsLock sync.Mutex
+	running        bool
+	config         *ControllerConfig
+)
```

Comment:
Not quite sure why these aren't in the struct?

Reply:
The variables in the struct don't have global scope, so they would have to be explicitly passed to functions when they are needed, or through other functions if they're needed in nested function calls.